Dockerfile

```diff
@@ -18,33 +26,41 @@ RUN echo "Built charon version=$(./charon version)"
 
 # Copy final binary into light stage.
 FROM debian:bookworm-slim
-RUN apt-get update && apt-get install -y ca-certificates wget fio
+RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates fio wget \
+  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ARG GITHUB_SHA=local
 ENV GITHUB_SHA=${GITHUB_SHA}
+
 COPY --from=builder /app/charon/charon /usr/local/bin/
+
 # Don't run container as root
 ENV USER=charon
 ENV UID=1000
 ENV GID=1000
-RUN addgroup --gid "$GID" "$USER"
-RUN adduser \
+RUN addgroup --gid "$GID" "$USER" \
+  && adduser \
     --disabled-password \
     --gecos "charon" \
     --home "/opt/$USER" \
     --ingroup "$USER" \
     --no-create-home \
     --uid "$UID" \
-    "$USER"
-RUN chown charon /usr/local/bin/charon
-RUN chmod u+x /usr/local/bin/charon
+    "$USER" \
+  && chown "$USER" /usr/local/bin/charon \
+  && chmod u+x /usr/local/bin/charon \
+  && mkdir -p "/opt/$USER" \
+  && chown "$USER" "/opt/$USER"
```

Comment:
Not necessary, as the `adduser` command already created this directory, see `--home "/opt/$USER"`. Its owner is also the said user.